### PR TITLE
[tf.learn] Allow Attention in RNNClassifier and Regressor

### PR DESCRIPTION
**Pull request by terrytangyuan**
1. Added attention wrappers in rnn model
2. Added test on additions and moved tests to a separate rnn_test.py
3. Allow attention params in rnn estimators

### CODE REVIEW

**Comment by martinwicke** [comment]
Without having checked -- the RNN interface has recently evolved a bit, did you use the new cell types (or can a user use those)?

**Comment by terrytangyuan** [comment]
@martinwicke Do you mean state_is_tuple change? If so, I specified a TODO there but haven't got a chance to look into differences yet.

**Comment by martinwicke** [comment]
I'll wait for @ilblackdragon to comment. Thanks!

File: tensorflow/contrib/learn/python/learn/estimators/rnn_test.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import tensorflow as tf
+from tensorflow.contrib.learn.python.learn.estimators._sklearn import accuracy_score
+from tensorflow.contrib.learn.python.learn.estimators._sklearn import mean_squared_error
+
+
+def rnn_input_fn(x):
+  return tf.split(1, 5, x)
+
+data = np.array(
+    list([[2, 1, 2, 2, 3], [2, 2, 3, 4, 5], [3, 3, 1, 2, 1], [2, 4, 5, 4, 1]
```
Comment (ilblackdragon) [inline tensorflow/contrib/learn/python/learn/estimators/rnn_test.py]: may be re-format it:
```[2, 1, 2..],
   [2, 2, 2...],
   ...
```

File: tensorflow/contrib/learn/python/learn/tests/nonlinear_test.py [hunk 2]
```diff
@@ -180,6 +107,5 @@ def _input_fn(x):
   #       9.74681854e-01, 1.19209290e-07]])
   #   self.assertAllClose(transformed, expected)
```
Comment (ilblackdragon) [inline tensorflow/contrib/learn/python/learn/tests/nonlinear_test.py]: two new lines is a standard.

File: tensorflow/contrib/learn/python/learn/estimators/rnn_test.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                                                          n_classes=2,
+                                                          input_op_fn=rnn_input_fn,
+                                                          bidirectional=True,
+                                                          attn_length=2,
+                                                          attn_size=2,
+                                                          attn_vec_size=2)
+    classifier.fit(data, labels)
+    predictions = classifier.predict(test_data)
+    self.assertAllClose(predictions, np.array([1, 0]))
+
```
Comment (ilblackdragon) [inline tensorflow/contrib/learn/python/learn/estimators/rnn_test.py]: two new lines between top level functions / classes / etc.

**Comment by ilblackdragon** [comment]
I actually never used this kind of attention, usually using attention in seq2seq models or attentive reader / memory network ones. Would be interesting to see how results are different between plain lstm/gru and this on some non-sentiment analysis (paper only mentioned sentiment analysis task).

**Comment by terrytangyuan** [comment]
@ilblackdragon I think I've addressed your comments. Thanks!

**Pull request merged**